### PR TITLE
obs-ffmpeg, UI: Allow passing muxer and protocol options for the ffmpeg-mpegts-muxer output.

### PR DESCRIPTION
**Pull request by pkviet**
### Description
For the rist or srt protocols, most protocol options can be passed through URL ( proto://IP:port?option1=val1&option2=val2 ).
But some can't and have to be passed programmatically through the private_data of an URLContext.
In addition muxer options can't be specified at the moment due to lack of UI.
This PR allows setting both muxer options and protocol options from Settings > Stream > Custom .
The syntax is the usual one:
ex: `option1=val1 option2=val2`
The options available are listed here:   
- mpegts muxer options : https://ffmpeg.org/ffmpeg-formats.html#toc-mpegts-1
- rist options : https://ffmpeg.org/ffmpeg-protocols.html#rist
- srt options : https://ffmpeg.org/ffmpeg-protocols.html#srt

For the UI, the choice has been made to put the new option text fields in the Stream Settings rather than in the Ouput Settings because the fields are meaningful only for the Custom stream service and only with mpegts streams (srt or rist in practice).
obs64_2021-11-04_13-15-44
These new fields are ignored for the rtmp protocol. 
obs64_2021-11-04_13-15-39
Tooltips explain the syntax for the options.
obs64_2021-11-04_13-41-07

### Motivation and Context
The motivation is that it can sometimes be necessary to specify some of these muxer or protocol options.
An example is that it is impossible otherwise to stream to Amazon Elemental Media Connect with rist.
Media Connect allows only rist simple profile while the default profile is main in avformat for rist.
With the enhancements brought by this PR, rist streaming from obs to Media Connect works though.

### How Has This Been Tested?
Tested by streaming with rist and srt to Elemental Media Connect, to vlc (in simple profile for rist).
(build done on vs2019 on win 10)

### Types of changes
 - New feature (non-breaking change which adds functionality) 

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/data/locale/en-US.ini [hunk 1]
```diff
@@ -829,6 +829,11 @@ Basic.Settings.Stream.Recommended.MaxVideoBitrate="Maximum Video Bitrate: %1 kbp
 Basic.Settings.Stream.Recommended.MaxAudioBitrate="Maximum Audio Bitrate: %1 kbps"
 Basic.Settings.Stream.Recommended.MaxResolution="Maximum Resolution: %1"
 Basic.Settings.Stream.Recommended.MaxFPS="Maximum FPS: %1"
+Basic.Settings.Stream.Custom.MpegtsOptions="Use options for mpegts streams (srt, rist, udp)"
+Basic.Settings.Stream.Custom.MpegtsMuxerOptions="Mpeg-ts options"
+Basic.Settings.Stream.Custom.MpegtsMuxerOptions.TT="Syntax: option1=value1 option2=value2; refer to https://ffmpeg.org/ffmpeg-formats.html#toc-mpegts-1 for a list of options"
+Basic.Settings.Stream.Custom.MpegtsProtocolOptions="Protocol options"
+Basic.Settings.Stream.Custom.MpegtsProtocolOptions.TT="Syntax: option1=value1 option2=value2; refer to https://ffmpeg.org/ffmpeg-protocols.html for a list of options"
```
Comment (WizardCM) [inline UI/data/locale/en-US.ini:836]:
```suggestion
Basic.Settings.Stream.Custom.MpegtsOptions="Configure parameters for MPEG-TS streams (SRT, RIST, UDP)"
Basic.Settings.Stream.Custom.MpegtsMuxerOptions="MPEG-TS options"
Basic.Settings.Stream.Custom.MpegtsMuxerOptions.TT="Syntax: option1=value1 option2=value2\nRefer to https://ffmpeg.org/ffmpeg-formats.html#toc-mpegts-1 for a list of options"
Basic.Settings.Stream.Custom.MpegtsProtocolOptions="Protocol options"
Basic.Settings.Stream.Custom.MpegtsProtocolOptions.TT="Syntax: option1=value1 option2=value2\Refer to https://ffmpeg.org/ffmpeg-protocols.html for a list of options"
```

**Comment by WizardCM** [comment]
Out of curiosity, are there certain defaults we could set for RIST behind the scenes rather than providing an advanced field that users could use incorrectly?

**Comment by pkviet** [comment]
For rist the main params are set through url. That's bandwidth and buffer.
The important protocol option which can't be passed through url and must be
passed programmatically is the rist profile (simple, main or advanced).
It could be a checkbox.
But it can't be set behind the back of the user. He/she has to set it
explicitly.
At least with the current version of librist.
I could remove the mpegts options and replace the protocol text widget by a
simple checkbox.

**Comment by pkviet** [comment]
on second thoughts, these option fields belong more to an advanced use. So we could decide to support only the rist main profile and not include these fields.
I'll close for now since even if we decide to have a checkbox for the rist profile, it'll require a UI rework.

**Pull request closed without merging**